pkg/auth/authenticator/password/ldappassword/url.go

```diff
@@ -0,0 +1,141 @@
+package ldappassword
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/go-ldap/ldap"
+)
+
+// LDAPURL holds a parsed RFC 2255 URL
+type LDAPURL struct {
+	// Scheme is ldap or ldaps
+	Scheme string
+	// HostPort is the host:port of the LDAP server
+	HostPort string
+	// The DN of the branch of the directory where all searches should start from
+	BaseDN string
+	// The attribute to search for
+	QueryAttribute string
+	// The scope of the search. Can be either 'one', 'sub', or 'base'
+	Scope int
+	// A valid LDAP search filter (e.g. "(objectClass=*)")
+	Filter string
+}
+
+// ParseURL parsed the given ldapURL as an RFC 2255 URL
+func ParseURL(ldapURL string) (LDAPURL, error) {
+	// Must be a valid URL to start
+	parsedURL, err := url.Parse(ldapURL)
+	if err != nil {
+		return LDAPURL{}, err
+	}
+
+	opts := LDAPURL{}
+
+	// Set scheme (default to ldap)
+	opts.Scheme = parsedURL.Scheme
+	switch opts.Scheme {
+	case "":
+		return LDAPURL{}, fmt.Errorf("scheme required")
+	case ProtocolLDAP, ProtocolLDAPS:
+		// ok
+	default:
+		return LDAPURL{}, fmt.Errorf("invalid scheme %q", parsedURL.Scheme)
+	}
+
+	// Set host (default to localhost to match mod_auth_ldap)
+	opts.HostPort = parsedURL.Host
+	if len(opts.HostPort) == 0 {
+		opts.HostPort = "localhost"
```

Comment:
This seems weird.  We're unwilling to default scheme to `ldaps`, but we will default `HostPort`?  Actually, why do we default this?